.github/workflows/test.yaml

```
@@ -19,7 +19,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: "1.21"
+          go-version: "1.22"
```

Could switch this to using `go-version-file` which will keep it using the same version as go.mod.
```suggestion
          go-version-file: 'go.mod'
```

https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file